apps/client/src/app/core/scheme/categories/categories.component.ts

```diff
@@ -0,0 +1,181 @@
+  fetchEditCategory: boolean = false
+  categoryEdit: formPayloadEdit = {
+    name: '',
+    helpText: '',
+    id: 0
+  }
+  closeDialog() {
+    this.visibleCreate = false
+    this.visibleEdit = false
+    this.formGroup.reset()
+    this.errors = {
+      name: '',
+      helpText: ''
+    }
+  }
+
+  showDialogEdit(category: Category) {
+    this.visibleEdit = true
+    this.categoryEdit = category
+    this.formGroup.controls.name.setValue(this.categoryEdit.name)
+    this.formGroup.controls.helpText.setValue(this.categoryEdit.helpText)
+  }
+  confirmationDelete(event: Event, id: number, name: string) {
+    this.confirmationService.confirm({
+      target: event.target as EventTarget,
+      message: `¿Estás seguro de que quieres eliminar la categoría <strong>${name}</strong>?`,
+      header: 'Eliminar categoría',
+      icon: 'pi pi-info-circle',
+      acceptButtonStyleClass: 'p-button-danger p-button-text',
+      rejectButtonStyleClass: 'p-button-text p-button-text',
+      acceptIcon: 'none',
+      rejectIcon: 'none',
+      acceptLabel: 'Sí',
+      rejectLabel: 'No',
+
+      accept: () => {
+        this.toast.show('info', 'Eliminando..', 'Eliminando criterio..')
+        this.onDelete(id)
+      },
+      reject: () => {
+        this.toast.show('error', 'Rechazado', 'Haz rechazado la eliminación')
+      }
+    })
+  }
+  getCategoriesPerIndicator(indicatorId: number) {
+    this.categoryService
+      .getAllCategoriesByIndicatorIndex(indicatorId)
+      .subscribe({
+        next: (res) => {
+          this.categories =
+            res.status === 'success' ? res.data : this.categories
+        },
+        error: (e) => {
+          console.error(e)
+          this.toast.show('error', 'Error', e.error.data.message)
+        }
+      })
+  }
+  onDelete(id: number) {
+    this.categoryService.delete(id).subscribe({
+      next: () => {
+        this.toast.show('success', 'Eliminado', 'Categoría eliminada con éxito')
+        this.getCategoriesPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        console.error(e)
+        this.toast.show('error', 'Error', e.error.data.message)
+      }
+    })
+  }
+  onCreate() {
+    this.fetchCreateCategory = true
+    const { helpText, name } = this.formGroup.controls
+    if (!name.value || !helpText.value) return
+    const category: CategoryDTO = {
+      indicatorIndex: this.indicatorIndex,
+      name: name.value,
+      helpText: helpText.value
+    }
+    this.categoryService.create(category).subscribe({
+      next: () => {
+        this.fetchCreateCategory = false
+        this.toast.show('success', 'Creado', 'Categoría creada con éxito')
+        this.closeDialog()
+        this.getCategoriesPerIndicator(this.indicatorIndex)
+      },
+      error: (e) => {
+        this.fetchCreateCategory = false
+        this.toast.show('error', 'Error', e.error.data.message)
+        this.closeDialog()
+      }
+    })
+  }
+  onEdit() {
```

Comment:
No estás chequeando que el form sea válido

```suggestion
  onEdit() {
    if (this.formGroup.invalid) return
    ...
```